packages/agents/lighthouse/src/tasks/processFromRoot/helpers/xlayer.ts

```diff
@@ -110,23 +112,113 @@ const getMerkleProof = async (apiUrl: string, depositCount: string, networkId: n
+  hubDomainId: number,
+  spokeDomainId: number,
+  _requestContext: RequestContext,
+): Promise<(RootMessage & Deposit) | undefined> => {
+  const {
+    logger,
+    config,
+    adapters: { contracts, chainreader },
+  } = getContext();
+  const { requestContext, methodContext } = createLoggingContext(getLatestXLayerSpokeMessage.name, _requestContext);
+  logger.info("Method start", requestContext, methodContext);
+
+  const xlayerBridgeApiEndpoint =
+    domainToChainId(hubDomainId) === 1
+      ? "https://rpc.xlayer.tech/priapi/v1/ob/bridge/"
+      : "https://testrpc.x1.tech/priapi/v1/ob/bridge";
+
+  const spokeConnector = contracts.spokeConnector(
+    domainToChainId(spokeDomainId),
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+  if (!spokeConnector) {
+    throw new NoSpokeConnector(spokeDomainId, requestContext, methodContext);
+  }
+
+  const hubConnector = contracts.hubConnector(
+    domainToChainId(hubDomainId),
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+  if (!hubConnector) {
+    throw new NoHubConnector(hubDomainId, requestContext, methodContext);
+  }
+
+  const spokeDeposits = await getDeposits(xlayerBridgeApiEndpoint, spokeConnector.address);
+  const [latest] = spokeDeposits.filter((d) => d.ready_for_claim).sort((a, b) => +b.block_num - +a.block_num);
+  if (!latest) {
+    return undefined;
+  }
+
+  // if the latest deposit is proven, return undefined
+  if (latest.claim_tx_hash) {
+    return undefined;
+  }
+
+  // get the transaction on L1
+  const tx = await chainreader.getTransactionReceipt(hubDomainId, latest.tx_hash);
+  // get the block on L1
+  const block = await chainreader.getBlock(hubDomainId, Number(tx.blockNumber));
+  if (!block || !tx) {
+    logger.warn("Failed to get block or tx for spoke message", requestContext, methodContext, {
+      hubDomainId,
+      block,
+      tx,
+      blockNumber: latest.block_num,
+      hash: latest.tx_hash,
+    });
+    return undefined;
+  }
+  // get the root
+  const iface = new Interface(hubConnector.abi as string[]);
+  const [parsed] = tx.logs
+    .filter((log) => log.address.toLowerCase() === hubConnector.address.toLowerCase())
+    .map((log) => iface.parseLog(log));
+
+  // return latest message
+  return {
+    ...latest,
+    id: `${parsed.args.data}-${spokeDomainId}`, // TODO: will this override existing carto entries?
+    spokeDomain: spokeDomainId.toString(),
+    hubDomain: hubDomainId.toString(),
+    root: parsed.args.data,
+    caller: tx.from.toLowerCase(),
+    transactionHash: latest.tx_hash.toLowerCase(),
+    timestamp: block.timestamp,
+    gasPrice: tx.effectiveGasPrice.toString(),
+    gasLimit: tx.cumulativeGasUsed.toString(),
+    blockNumber: block.number,
+    processed: false,
+    count: 0, // TODO: make more accurate
+    // only defined if claim tx hash is defined
+    sentTaskId: undefined,
+    sentTimestamp: undefined,
+    relayerType: undefined,
+  };
+};
```

Comment:
@preethamr need help with count default here

Reply:
`0` would work for l1-l2 xlayer specific roots. Nothing down the message pipe should be expecting this